siad/daemon.go

```diff
@@ -77,16 +77,32 @@ func processModules(modules string) (string, error) {
 	return modules, nil
 }
 
+// processProfiles checks that the flags given for profiling are valid
+func processProfile(profile string) (string, error) {
+	profile = strings.ToLower(profile)
+	validProfiles := "cmt"
+
+	invalidProfiles := profile
+	for _, p := range validProfiles {
+		invalidProfiles = strings.Replace(invalidProfiles, string(p), "", 1)
+	}
```

Comment:
That regular expression doesn't catch duplicate flags though. The code here is the same as was used in processModules in the same file also.